pkgs/misc/vim-plugins/update.py

```diff
@@ -111,6 +118,22 @@ def latest_commit(self) -> Tuple[str, datetime]:
             updated = datetime.strptime(updated_tag.text, "%Y-%m-%dT%H:%M:%SZ")
             return Path(str(url.path)).name, updated
 
+    def check_for_redirect(self, url: str, req: http.client.HTTPResponse):
+        response_url = req.geturl()
+        if url != response_url:
+            new_owner, new_name = (
+                urllib.parse.urlsplit(response_url).path.strip("/").split("/")[:2]
```

Comment:
This seems surprising to me. Why do we want the first two parts of the URL? I would assume we need the last two parts instead:

```
github.com/owner/repo
```

-> First two would be `github.com` as the owner and `owner` as the repo.

Am I missing something?

Reply:
https://docs.python.org/3/library/urllib.parse.html#module-urllib.parse

github.com is the `netloc` in this scenario and we're only looking at the `path`.